components/proxy/src/main/java/com/hotels/styx/BuiltInInterceptors.java

```diff
@@ -50,10 +50,14 @@ private BuiltInInterceptors() {
             builder.add(new HttpMessageLoggingInterceptor(longFormatEnabled, httpMessageFormatter));
         }
 
+        final HttpInterceptor viaInterceptor = config.get("via")
+                .map(ViaHeaderAppendingInterceptor::new)
+                .orElseGet(ViaHeaderAppendingInterceptor::new);
```

Comment:
AFAICS this adds a new `via` config option in the styx server configuration. Right?

In that case:

* It needs to be added to the `ServerConfigSchema` file that validates the configuration. Otherwise styx wouldn't recognise the new `via` option, and would refuse to start:

```
styx_1       | 11:07:53.040 [main] ERROR com.hotels.styx.StyxServer - Styx server failed to start due to configuration error in file [/styx/config/styxconf.yml]: Unexpected field: 'via'
```

 If you don't see this error, you may have disabled the config validation with `validateServerConfig` system property.

* Rather than adding the new `via` option in the configuration root, it should go under the `proxy` server settings. This way, when we'll start properly supporting multiple server ports, the via header could be configured separately for each connector. I would add it here:

```
        STYX_SERVER_CONFIGURATION_SCHEMA_BUILDER = newDocument()
                    .rootSchema(object(
                            optional("proxy", object(
                                    optional("compressResponses", bool()),

                                    ...

                                    optional("via", string())    // <-- here
                            )),
```

* In fact, styx already supports multiple HTTP servers. Check the `StyxHttpServer.kt` file. When this feature becomes de-facto way of starting styx servers, the StyxHttpServer configuration block would be the only way to configure the new `via` header.

Reply:
> AFAICS this adds a new `via` config option in the styx server configuration. Right?
> 

Yes, this is correct.

> * It needs to be added to the `ServerConfigSchema` file that validates the configuration. Otherwise styx wouldn't recognise the new `via` option, and would refuse to start:
> 
> ```
> styx_1       | 11:07:53.040 [main] ERROR com.hotels.styx.StyxServer - Styx server failed to start due to configuration error in file [/styx/config/styxconf.yml]: Unexpected field: 'via'
> ```
> 
> If you don't see this error, you may have disabled the config validation with `validateServerConfig` system property.

Will do.

> * Rather than adding the new `via` option in the configuration root, it should go under the `proxy` server settings. This way, when we'll start properly supporting multiple server ports, the via header could be configured separately for each connector.

This makes sense, and I almost put it there to begin with, but opted to defer to review. I will move it.